src/main/kotlin/de/cyface/collector/handler/PreRequestHandler.kt

```diff
@@ -40,67 +40,73 @@ import java.net.URL
 import java.util.Locale
 
 /**
- * A handler for receiving HTTP POST requests on the "measurements" end point.
+ * A handler for receiving HTTP POST requests on the "measurements" or "attachments" end point.
  * This end point tells the client if the upload may continue or should be skipped.
  *
  * @author Armin Schnabel
  * @author Klemens Muthmann
- * @version 2.0.0
- * @since 6.0.0
  * @property storageService The service used to store data and retrieve information about stored data
  *                          by this application.
- * @property measurementLimit The maximal number of `Byte`s which may be uploaded in the upload request.
+ * @property uploadLimit The maximal number of `Byte`s which may be uploaded in the upload request.
  * @property httpPath The path of the URL under which the Collector service is deployed. This is used to return the
  * correct "Location" header.
  */
 class PreRequestHandler(
     private val storageService: DataStorageService,
-    private val measurementLimit: Long,
+    private val uploadLimit: Long,
     private val httpPath: String
 ) : Handler<RoutingContext> {
 
+    private lateinit var uploadStrategy: UploadStrategy
+
     override fun handle(ctx: RoutingContext) {
         LOGGER.info("Received new pre-request.")
         val request = ctx.request()
         val session = ctx.session()
         val metaData = ctx.body().asJsonObject()
         try {
-            bodySize(request.headers(), measurementLimit, X_UPLOAD_CONTENT_LENGTH_FIELD)
+            // Decide which upload strategy to use
+            uploadStrategy = if (request.path().contains("attachments")) {
```

Comment:
This is ugly. There should be two handlers for each case registered at the appropriate endpoints on the Vert.x Router.

Reply:
Additionally, this is no proper strategy, as the object is not injected and thus impossible to mock or switch especially during testing.